collector/serverstats.go

```diff
@@ -0,0 +1,186 @@
+			nil,
+			nil,
+		),
+		prometheus.CounterValue,
+	}
+
+	serverstatsNTPDrops = typedDesc{
+		prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, serverstatsSubsystem, "ntp_packets_dropped_total"),
+			"The number of NTP requests dropped by the server due to rate limiting.",
+			nil,
+			nil,
+		),
+		prometheus.CounterValue,
+	}
+
+	serverstatsNKEDrops = typedDesc{
+		prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, serverstatsSubsystem, "nts-ke_connections_dropped_total"),
+			"The number of NTS-KE connections dropped by the server due to rate limiting.",
+			nil,
+			nil,
+		),
+		prometheus.CounterValue,
+	}
+
+	serverstatsCMDDrops = typedDesc{
+		prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, serverstatsSubsystem, "command_packets_dropped_total"),
+			"The number of command requests dropped by the server due to rate limiting.",
+			nil,
+			nil,
+		),
+		prometheus.CounterValue,
+	}
+
+	serverstatsLogDrops = typedDesc{
+		prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, serverstatsSubsystem, "client_log_records_dropped_total"),
+			"The number of client log records dropped by the server to limit the memory use.",
+			nil,
+			nil,
+		),
+		prometheus.CounterValue,
+	}
+
+	serverstatsNTPAuthHits = typedDesc{
+		prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, serverstatsSubsystem, "auhtenticated_ntp_packets_total"),
+			"The number of received NTP requests that were authenticated (with a symmetric key or NTS).",
+			nil,
+			nil,
+		),
+		prometheus.CounterValue,
+	}
+
+	serverstatsNTPInterleavedHits = typedDesc{
+		prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, serverstatsSubsystem, "interleaved_ntp_packets_total"),
+			"The number of received NTP requests that were detected to be in the interleaved mode.",
+			nil,
+			nil,
+		),
+		prometheus.CounterValue,
+	}
+
+	serverstatsNTPTimestamps = typedDesc{
+		prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, serverstatsSubsystem, "ntp_timestamps_held"),
+			"The number of pairs of receive and transmit timestamps that the server is currently holding in memory for clients using the interleaved mode.",
+			nil,
+			nil,
+		),
+		prometheus.GaugeValue,
+	}
+
+	serverstatsNTPSpanSeconds = typedDesc{
+		prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, serverstatsSubsystem, "ntp_timestamp_span"),
+			"The interval (in seconds) covered by the currently held NTP timestamps.",
+			nil,
+			nil,
+		),
+		prometheus.GaugeValue,
+	}
+)
+
+func (e Exporter) getServerstatsMetrics(ch chan<- prometheus.Metric, client chrony.Client) error {
+	packet, err := client.Communicate(chrony.NewServerStatsPacket())
+	if err != nil {
+		return err
+	}
+	level.Debug(e.logger).Log("msg", "Got 'serverstats' response", "serverstats_packet", packet.GetStatus())
+
+	serverstats, ok := packet.(*chrony.ReplyServerStats3)
+	if !ok {
+		return fmt.Errorf("Got wrong 'serverstats' response: %q", packet)
```

Comment:
Go style nit

```suggestion
		return fmt.Errorf("got wrong 'serverstats' response: %q", packet)
```

Reply:
The version with the big G is consistent with the other collectors, should those be changed too?

Reply:
I can take care of fixing the other ones. I just noticed this in the linter.